src/main/java/mari/moita/com/Exercise4.java

```diff
@@ -0,0 +1,43 @@
+package mari.moita.com;
+
+import java.util.List;
+
+public class Exercise4 {
+
+    public void listContainsManAndWoman(List<Person> people) {
+
+        boolean female = false;
+        boolean male = false;
+
+        for (Person p : people) {
+            if ("male".equals(p.getGender())) {
+                male = true;
+            } else if ("female".equals(p.getGender())) {
+                female = true;
+            }
+        }
+        if (!female || !male) {
+            throw new IllegalArgumentException("List must have, at least, one male and one female.");
+        }
+    }
+
+    public Person[] OldestManAndYoungestWoman(List<Person> people) {
```

Comment:
the method name could be something that indicates that it will return both oldest and youngest people, someethiing like:
- getOldestManAndYoungestWoman
- extractOldestManAndYoungestWoman
- findOldestManAndYoungestWoman